web/src/settings_display.js

```diff
@@ -100,21 +100,25 @@ function user_default_language_modal_post_render() {
                 setting_value,
             );
 
-            change_display_setting(
+            const $spinner = $("#settings_content").find(".general-settings-status");
+            $spinner.fadeTo(0, 1);
+            loading.make_indicator($spinner, {text: $t({defaultMessage: "Saving"})});
+            channel.patch({
+                url: "/json/settings",
                 data,
-                $("#settings_content").find(".general-settings-status"),
-                $t_html(
-                    {
-                        defaultMessage:
-                            "Saved. Please <z-link>reload</z-link> for the change to take effect.",
-                    },
-                    {
-                        "z-link": (content_html) =>
-                            `<a class='reload_link'>${content_html.join("")}</a>`,
-                    },
-                ),
-                true,
-            );
+                success(response_data) {
+                    const appear_after = 500;
+                    const success_msg_html =
+                        response_data.msg || $t_html({defaultMessage: "Saved"});
```

Comment:
Is there any need to add  the "||" condition here? We would always expect a message returned by server.